packages/adapter-chiselstrike/src/index.ts

```diff
@@ -0,0 +1,185 @@
+            resp = await fetcher.fetch(fetcher.accounts(`?.userId=${userId}`), { method: 'DELETE' })
+            await ensureOK(resp, `deleting accounts for ${userId}`)
+            return null
+        },
+        // linkAccount: (account: Account) => Promise<void> | Awaitable<Account | null | undefined>;
+        linkAccount: async (account: Account): Promise<Account | null | undefined> => {
+            const body = JSON.stringify(account);
+            const resp = await fetcher.fetch(fetcher.accounts(), { method: 'POST', body })
+            await ensureOK(resp, `posting account ${body}`);
+            return await resp.json()
+        },
+        // unlinkAccount?: ((providerAccountId: Pick<Account, "provider" | "providerAccountId">) => Promise<void> | Awaitable<Account | undefined>) | undefined;
+        unlinkAccount: async (providerAccountId: Pick<Account, "provider" | "providerAccountId">): Promise<void> => {
+            const resp = await fetcher.fetch(fetcher.accounts(`?${makeFilter(providerAccountId)}`), { method: 'DELETE' })
+            await ensureOK(resp, `deleting account ${JSON.stringify(providerAccountId)}`)
+        },
+        // createSession: (session: { sessionToken: string; userId: string; expires: Date; }) => Awaitable<AdapterSession>;
+        createSession: async (session: { sessionToken: string; userId: string; expires: Date; }): Promise<AdapterSession> => {
+            const str = JSON.stringify(session);
+            const resp = await fetcher.fetch(fetcher.sessions(), { method: 'POST', body: str })
+            await ensureOK(resp, `posting session ${JSON.stringify(session)}`);
+            return sessionFromJson(await resp.json())
+        },
+        // getSessionAndUser: (sessionToken: string) => Awaitable<{ session: AdapterSession; user: AdapterUser; } | null>;
+        getSessionAndUser: async (sessionToken: string): Promise<{ session: AdapterSession; user: AdapterUser; } | null> => {
+            const session = await fetcher.filter(fetcher.sessions(`?.sessionToken=${sessionToken}`))
+            if (!session) { return null }
+            const rUser = await fetcher.fetch(fetcher.users(`/${session.userId}`))
+            await ensureOK(rUser, `fetching user ${session.userId}`)
+            return { session: sessionFromJson(session), user: userFromJson(await rUser.json()) }
+        },
+        // updateSession: (session: Partial<AdapterSession> & Pick<AdapterSession, "sessionToken">) => Awaitable<AdapterSession | null | undefined>;
+        updateSession: async (session: Partial<AdapterSession> & Pick<AdapterSession, "sessionToken">): Promise<AdapterSession | null | undefined> => {
+            // TODO: use PATCH
+            const dbSession = await fetcher.filter(fetcher.sessions(`?.sessionToken=${session.sessionToken}`))
+            if (!dbSession) { return null }
+            Object.keys(session).forEach(key => dbSession[key] = session[key])
```

Comment:
type error here. `AdapterSession` has a field `expires: Date` but `dbSession: { [_: string]: string }`. So you can't simply do `dbSession[key] = session[key]`
```suggestion
            Object.entries(session).forEach(([key, entry]) => {
                if (entry instanceof Date) {
                    dbSession[key] = entry.toISOString();
                } else {
                    dbSession[key] = entry;
                }
           });
```